src/test/java/uk/gov/pay/products/service/PaymentFinderTest.java

```diff
@@ -64,12 +74,74 @@ public void shouldReturnAList_whenFoundByProductExternalId() throws Exception{
         paymentEntity.setExternalId(externalPaymentId);
         List<PaymentEntity> paymentList = Arrays.asList(paymentEntity);
 
-       when(paymentDao.findByProductExternalId(externalProductId)).thenReturn(paymentList);
+        when(paymentDao.findByProductExternalId(externalProductId)).thenReturn(paymentList);
 
         List<Payment> expectedPaymentList = paymentFinder.findByProductExternalId(externalProductId);
 
         assertThat(expectedPaymentList.isEmpty(), is(false));
         assertThat(expectedPaymentList.get(0).getExternalId(), is(externalPaymentId));
         assertThat(expectedPaymentList.size(), is(1));
     }
+
+    @Test
+    public void shouldReturnPaymentWithExternalStatus_whenFoundByExternalIdAndStatusIsSubmitted() throws Exception {
+        String externalId = randomUuid();
+        PaymentEntity paymentEntity = new PaymentEntity();
+        ProductEntity productEntity = ProductEntityFixture.aProductEntity().build();
+        paymentEntity.setExternalId(externalId);
+        paymentEntity.setProductEntity(productEntity);
+        paymentEntity.setStatus(PaymentStatus.SUBMITTED);
+        paymentEntity.setGovukPaymentId(externalId);
+        PaymentResponse paymentResponse = new PaymentResponse();
+        paymentResponse.setState(new PaymentState("submitted", true, "a message", "a code"));
+        when(paymentDao.findByExternalId(externalId)).thenReturn(Optional.of(paymentEntity));
+        when(publicApiRestClient.getPayment(productEntity.getPayApiToken(), paymentEntity.getGovukPaymentId()))
+                .thenReturn(Optional.of(paymentResponse));
+
+        Optional<Payment> optionalPayment = paymentFinder.findByExternalIdWithExternalStatus(externalId);
+        assertThat(optionalPayment.isPresent(), is(true));
+
+        Payment payment = optionalPayment.get();
+        assertThat(payment.getGovUkStatus(), is("submitted"));
+    }
+
+    @Test
+    public void shouldReturnPaymentWithNoExternalStatus_whenFoundByExternalIdAndStatusIsCreated() throws Exception {
+        String externalId = randomUuid();
+        PaymentEntity paymentEntity = new PaymentEntity();
+        ProductEntity productEntity = ProductEntityFixture.aProductEntity().build();
+        paymentEntity.setExternalId(externalId);
+        paymentEntity.setProductEntity(productEntity);
+        paymentEntity.setStatus(PaymentStatus.CREATED);
+        paymentEntity.setGovukPaymentId(externalId);
+        PaymentResponse paymentResponse = new PaymentResponse();
+        paymentResponse.setState(new PaymentState("submitted", true, "a message", "a code"));
```

Comment:
Do we need a `paymentResponse` for this test?

Reply:
👍